File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/dao/impl/OrderDAOImpl.java
package com.firefly.example.reactive.coffee.store.dao.impl;

import com.firefly.annotation.Component;
import com.firefly.example.reactive.coffee.store.dao.OrderDAO;
import com.firefly.example.reactive.coffee.store.model.Order;
import com.firefly.reactive.adapter.db.ReactiveSQLConnection;
import reactor.core.publisher.Mono;

import java.util.List;

/**
 * @author Pengtao Qiu
 */
@Component
public class OrderDAOImpl implements OrderDAO {

    @Override
    public Mono<List<Long>> insertBatch(List<Order> list, ReactiveSQLConnection connection) {
        return connection.inTransaction(c -> c.insertObjectBatch(list, Order.class));
    }
}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/dao/impl/OrderDAOImpl.kt
package com.firefly.example.kotlin.coffee.store.dao.impl

import com.firefly.annotation.Component
import com.firefly.annotation.Inject
import com.firefly.example.kotlin.coffee.store.dao.OrderDAO
import com.firefly.example.kotlin.coffee.store.model.Order
import com.firefly.kotlin.ext.db.AsyncTransactionalManager
import com.firefly.kotlin.ext.db.asyncInsertObjectBatch

/**
 * @author Pengtao Qiu
 */
@Component
class OrderDAOImpl : OrderDAO {

    @Inject
    private lateinit var db: AsyncTransactionalManager

    override suspend fun insertBatch(orders: List<Order>): List<Long> = db.execSQL {
        it.asyncInsertObjectBatch<Order, Long>(orders)
    }

}